### ISSUE TITLE
Abstract Elasticsearch access for supporting different versions

### ISSUE THREAD

**Issue by fehmicansaglam** [fehmicansaglam/esctl#4]
### Description:
Currently, the esctl tool is tightly coupled to a specific Elasticsearch version, limiting its flexibility. The goal is to abstract the Elasticsearch access in the es package, allowing esctl to seamlessly support different Elasticsearch versions. This will enhance flexibility and maintainability.

### Acceptance Criteria:

- Abstract the Elasticsearch access in the es package.
- Implement version-specific modules for different Elasticsearch versions.
- Refactor esctl to utilize the abstracted Elasticsearch access.
- Ensure compatibility and functionality across supported Elasticsearch versions.

**Comment by hkulekci** [comment]
Would it also be beneficial to consider supporting OpenSearch while working on this abstraction?

**Comment by fehmicansaglam** [comment]
Makes sense. I am not aware of any differences between Elasticsearch and OpenSearch management APIs though. Have you noticed any discrepancies?

**Comment by hkulekci** [comment]
I could not check this tool with OpenSearch yet. Let me do a test and share with you.

**Comment by hkulekci** [comment]
I just checked for some features, and it is working as well. Just for the local setup, SSL verification needs to be skipped: 

https://github.com/hkulekci/esctl/tree/skip-ssl-verification
